dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ThreadPoolController.java

```diff
@@ -832,29 +866,42 @@ boolean handleOutliers(ThroughputDistribution distribution, double throughput) {
             Integer priorKey = Integer.valueOf(poolSize);
             int smallerPools = 0;
             while (true) {
-                int distance = 0;
                 // stop if we run out of data
                 if (shrinkKey == null)
                     break;
                 priorKey = shrinkKey;
                 shrinkKey = threadStats.lowerKey(shrinkKey);
+                int distance = 0;
+                boolean pruned = false;
+                boolean inHangBuffer = true;
                 // discard invalid data (old or out-of-range) found in comparison stats
                 ThroughputDistribution priorStats = threadStats.get(priorKey);
                 distance = poolSize - priorKey.intValue();
                 if (pruneData(priorStats, forecast, distance, downwardCompareSpan)) {
-                    threadStats.remove(priorKey);
-                } else {
+                    if (priorKey > hangBufferPoolSize) {
+                        threadStats.remove(priorKey);
+                        pruned = true;
+                        inHangBuffer = false;
+                    }
+                }
+                if (!pruned) {
                     // found a valid smaller poolSize, so include that data in the shrinkScore
                     smallerPools++;
                     shrinkScore += priorStats.getProbabilityGreaterThan(forecast);
-                    // stop if reached/passed compareSpan
-                    if (distance >= downwardCompareSpan)
-                        break;
+                    if (inHangBuffer) {
+                        // stop when we get compareRange datapoints
+                        if (smallerPools >= compareRange)
+                            break;
+                    } else {
+                        // stop if reached/passed compareSpan
+                        if (distance >= downwardCompareSpan)
```

Comment:
Can we just make it an else if?

Reply:
Does it matter? 

The way it looks on the screen is clear to me now, and it surely functions as an else if ...

Reply:
Code style thing again.  It is the same and doesn't matter that much.